### ISSUE TITLE
Different schema API

### ISSUE THREAD

**Issue by iafisher** [iafisher/isqlite#22]
Current API:

```python
schema = Schema([Table(...), Table(...)])

with Database("...") as db:
    schema.create(db)
    ...
    schema.add_column(db, ...)
```

Proposed API:

```python
with Database("...", tables=[Table(...), Table(...)]) as db:
    db.create_tables()
    ...
    db.add_column(...)
```

The `tables` parameter to the `Database` constructor would be optional, but certain operations (like `create_tables`) would fail if `tables` was not provided.

Alternatively, each schema-affecting method could simply take the table(s) as a parameter:

```python
table1 = Table(...)
table2 = Table(...)

with Database("...") as db:
    db.create_tables([table1, table2])
    ...
    db.add_column(table1, ...)
```

This could be confusing though because `Database.add_column` and the other methods wouldn't mutate the `Table` object as you might expect.

**Comment by iafisher** [comment]
Django's ORM can automatically figure out schema migrations based on changes to your Python models. I think isqlite should be lower-level than that, i.e. it should provide methods to add, remove, alter and reorder columns, but not magically do anything for you.

- `add_table` requires a table name and a list of column definitions.
- `drop_table` requires a table name.
- `add_column` requires a table name and a column definition.
- `drop_column`, `reorder_columns` and `alter_column` all require a table name and a list of the names of all the table's existing columns.

It should actually be possible to get a list of the table's existing columns by querying the `sqlite_master` table and using [python-sqlparse](https://sqlparse.readthedocs.io/en/latest/) to parse the result. So I don't think I need the concept of a schema object at all - individual methods can just accept column objects, and the user is free to maintain a schema as a Python data structure or not.

e.g., for a web app, they could have a `migrations` folder with `create_db.py` and `migrate_20210101.py`, `migrate_20210117.py`, etc. scripts and configure their deployment script to automatically run the migration scripts, skipping those already run. (Django uses a `django_migrations` table to track what migrations have been run on the database, but the user can handle it however they like. It's not isqlite's problem.)

**Comment by iafisher** [comment]
This has been mostly implemented in f04384c, but I'll keep this bug open for a bit until I decide I like the new API enough to keep.
